service/handlers.go

```diff
@@ -20,12 +21,22 @@ func createHealthcheckHandler(service *ProxyService) func(http.ResponseWriter, *
 
 		// check that the database is reachable
 		err := service.Database.HealthCheck()
-
 		if err != nil {
 			errMsg := fmt.Errorf("proxy service unable to connect to database")
 			combinedErrors = errors.Join(combinedErrors, errMsg)
 		}
 
+		// check that the cache is reachable
+		err = service.Cache.Healthcheck(context.Background())
```

Comment:
once you've updated being able to turn the cache on or off, this will only need to be checked when the flag is set for caching to be enabled

Reply:
make sense, done